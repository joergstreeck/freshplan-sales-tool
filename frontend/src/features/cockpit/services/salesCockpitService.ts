/**
 * Service f체r Sales Cockpit BFF-Integration
 *
 * Ruft aggregierte Dashboard-Daten vom Backend-for-Frontend ab
 */
import { httpClient } from '../../../lib/apiClient';
import type { SalesCockpitDashboard } from '../types/salesCockpit';

class SalesCockpitService {
  /**
   * L채dt alle Dashboard-Daten f체r einen bestimmten Benutzer
   *
   * @param userId Die ID des Benutzers
   * @returns Aggregierte Dashboard-Daten
   */
  async getDashboardData(userId: string): Promise<SalesCockpitDashboard> {
    const response = await httpClient.get<SalesCockpitDashboard>(
      `/api/sales-cockpit/dashboard/${userId}`
    );
    return response.data;
  }

  /**
   * Health-Check f체r den Sales Cockpit Service
   *
   * @returns Status des Services
   */
  async checkHealth(): Promise<{ status: string; service: string }> {
    const response = await httpClient.get<{ status: string; service: string }>(
      '/api/sales-cockpit/health'
    );
    return response.data;
  }
}

// Singleton-Instanz
export const salesCockpitService = new SalesCockpitService();
