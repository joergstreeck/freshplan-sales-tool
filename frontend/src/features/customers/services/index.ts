/**
 * Customer Services Export
 *
 * Zentrale Exportstelle f√ºr alle Customer Management Services.
 *
 * @see /Users/joergstreeck/freshplan-sales-tool/docs/features/FC-005-CUSTOMER-MANAGEMENT/03-FRONTEND/04-api-integration.md
 */

// API Clients
export { apiClient } from './api-client';
export { customerApi } from './customerApi';
export { fieldDefinitionApi } from './fieldDefinitionApi';
export { locationApi } from './locationApi';

// React Query Hooks
export {
  // Query Keys
  queryKeys,

  // Customer Hooks
  useCreateCustomerDraft,
  useCustomerDraft,
  useUpdateCustomerDraft,
  useFinalizeCustomerDraft,
  useSearchCustomers,
  useCustomer,
  useUpdateCustomer,
  useRecentCustomers,
  useCustomerStatistics,

  // Location Hooks
  useCustomerLocations,
  useCreateLocation,
  useUpdateLocation,

  // Field Definition Hooks
  useFieldDefinitions,
  useIndustries,
  usePreloadFieldDefinitions,
} from './hooks';

// Re-export types for convenience
export type { RequestConfig } from './api-client';
