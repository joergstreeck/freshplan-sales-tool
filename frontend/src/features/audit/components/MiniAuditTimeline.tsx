/**
 * Mini Audit Timeline Component
 * 
 * Compact audit history display for integration in contact cards.
 * Shows last change in collapsed state, expands to show last 5 changes.
 * 
 * @module MiniAuditTimeline
 * @since FC-005 PR4
 */

import React, { useState, useMemo } from 'react';
import {
  Box,
  Accordion,
  AccordionSummary,
  AccordionDetails,
  Typography,
  Chip,
  Stack,
  Skeleton,
  Alert,
  IconButton as _IconButton,
  Tooltip,
  Button,
  useTheme,
  alpha,
} from '@mui/material';
import {
  Timeline,
  TimelineItem,
  TimelineSeparator,
  TimelineDot,
  TimelineConnector,
  TimelineContent,
  TimelineOppositeContent,
} from '@mui/lab';
import {
  ExpandMore as ExpandMoreIcon,
  History as HistoryIcon,
  Edit as EditIcon,
  Add as AddIcon,
  Delete as DeleteIcon,
  Visibility as ViewIcon,
  Person as PersonIcon,
  Schedule as TimeIcon,
  Info as InfoIcon,
  Email as EmailIcon,
  Phone as PhoneIcon,
  Business as BusinessIcon,
  LocationOn as LocationIcon,
} from '@mui/icons-material';
import { formatDistanceToNow, format } from 'date-fns';
import { de } from 'date-fns/locale';
import { useQuery } from '@tanstack/react-query';
import { auditApi } from '../services/auditApi';
import type { AuditEntry } from '../types/audit.types';
import { useAuth } from '../../../contexts/AuthContext';

interface MiniAuditTimelineProps {
  entityType: string;
  entityId: string;
  maxEntries?: number;
  showDetails?: boolean;
  compact?: boolean;
  onShowMore?: () => void;
}

/**
 * Compact audit timeline for embedding in cards
 */
export function MiniAuditTimeline({
  entityType,
  entityId,
  maxEntries = 5,
  showDetails = false,
  compact = true,
  onShowMore,
}: MiniAuditTimelineProps) {
  const theme = useTheme();
  const { user } = useAuth();
  const [expanded, setExpanded] = useState(false);
  
  // Props are now validated through TypeScript
  
  // Check if user has permission to view audit
  // In development with authBypass, always allow audit viewing
  const isDevelopment = import.meta.env.DEV;
  const canViewAudit = isDevelopment || user?.roles?.some(role => 
    ['admin', 'manager', 'auditor'].includes(role)
  );
  
  // Permissions checked via role-based access
  
  // Fetch audit data
  const { data: auditEntries, isLoading, error } = useQuery({
    queryKey: ['audit', entityType, entityId, maxEntries],
    queryFn: async () => {
      // Fetching audit trail with proper entity identification
      const result = await auditApi.getEntityAuditTrail(entityType, entityId, 0, maxEntries);
      console.log('Audit trail result:', result);
      return result;
    },
    staleTime: 5 * 60 * 1000, // 5 minutes
    gcTime: 10 * 60 * 1000, // 10 minutes cache
    enabled: canViewAudit && !!entityId, // Also check that entityId exists
  });
  
  // Get last change summary
  const lastChange = useMemo(() => {
    if (!auditEntries?.content?.length) return null;
    const latest = auditEntries.content[0];
    return {
      user: latest.userName || 'System',
      time: formatDistanceToNow(new Date(latest.timestamp), {
        addSuffix: true,
        locale: de,
      }),
      action: latest.eventType,
    };
  }, [auditEntries]);
  
  // Action type to icon mapping
  const getActionIcon = (action: string) => {
    const iconMap: Record<string, JSX.Element> = {
      'CREATE': <AddIcon fontSize="small" sx={{ color: theme.palette.success.main }} />,
      'UPDATE': <EditIcon fontSize="small" sx={{ color: theme.palette.info.main }} />,
      'DELETE': <DeleteIcon fontSize="small" sx={{ color: theme.palette.error.main }} />,
      'VIEW': <ViewIcon fontSize="small" sx={{ color: theme.palette.grey[500] }} />,
      'CUSTOMER_CREATED': <AddIcon fontSize="small" sx={{ color: theme.palette.success.main }} />,
      'CUSTOMER_UPDATED': <EditIcon fontSize="small" sx={{ color: theme.palette.info.main }} />,
      'CONTACT_ADDED': <PersonIcon fontSize="small" sx={{ color: theme.palette.success.main }} />,
      'CONTACT_UPDATED': <PersonIcon fontSize="small" sx={{ color: theme.palette.info.main }} />,
      'CONTACT_DELETED': <PersonIcon fontSize="small" sx={{ color: theme.palette.error.main }} />,
      'EMAIL_SENT': <EmailIcon fontSize="small" sx={{ color: theme.palette.info.main }} />,
      'PHONE_CALL': <PhoneIcon fontSize="small" sx={{ color: theme.palette.info.main }} />,
      'ADDRESS_CHANGED': <LocationIcon fontSize="small" sx={{ color: theme.palette.warning.main }} />,
      'STATUS_CHANGED': <BusinessIcon fontSize="small" sx={{ color: theme.palette.warning.main }} />,
    };
    return iconMap[action] || <HistoryIcon fontSize="small" />;
  };
  
  // Action type to color mapping
  const getActionColor = (action: string): 'success' | 'info' | 'warning' | 'error' | 'default' => {
    if (action.includes('CREATE') || action.includes('ADD')) return 'success';
    if (action.includes('UPDATE') || action.includes('SENT') || action.includes('CALL')) return 'info';
    if (action.includes('DELETE') || action.includes('REMOVE')) return 'error';
    if (action.includes('WARNING') || action.includes('CHANGED')) return 'warning';
    return 'default';
  };
  
  // Format action label
  const formatActionLabel = (action: string): string => {
    const labelMap: Record<string, string> = {
      'CREATE': 'Erstellt',
      'UPDATE': 'Aktualisiert',
      'DELETE': 'Gelöscht',
      'VIEW': 'Angesehen',
      'CUSTOMER_CREATED': 'Kunde erstellt',
      'CUSTOMER_UPDATED': 'Kunde aktualisiert',
      'CONTACT_ADDED': 'Kontakt hinzugefügt',
      'CONTACT_UPDATED': 'Kontakt aktualisiert',
      'CONTACT_DELETED': 'Kontakt gelöscht',
      'EMAIL_SENT': 'E-Mail gesendet',
      'PHONE_CALL': 'Anruf getätigt',
      'ADDRESS_CHANGED': 'Adresse geändert',
      'STATUS_CHANGED': 'Status geändert',
    };
    return labelMap[action] || action;
  };
  
  // Format change details
  const formatChangeDetails = (changes: any): string => {
    if (!changes) return '';
    
    try {
      const changeObj = typeof changes === 'string' ? JSON.parse(changes) : changes;
      const fields = Object.keys(changeObj);
      
      if (fields.length === 0) return '';
      if (fields.length === 1) {
        const field = fields[0];
        const change = changeObj[field];
        if (change.from && change.to) {
          return `${field}: ${change.from} → ${change.to}`;
        }
        return `${field} geändert`;
      }
      
      return `${fields.length} Felder geändert`;
    } catch {
      return '';
    }
  };
  
  // Don't render if user doesn't have permission
  if (!canViewAudit) {
    return null;
  }
  
  // Loading state
  if (isLoading) {
    return (
      <Box sx={{ p: 1 }}>
        <Skeleton variant="text" width="80%" height={20} />
      </Box>
    );
  }
  
  // Error state
  if (error) {
    return (
      <Alert severity="error" sx={{ py: 0.5, fontSize: '0.75rem' }}>
        Audit-Historie konnte nicht geladen werden
      </Alert>
    );
  }
  
  // No data state
  if (!auditEntries?.content?.length) {
    return (
      <Typography variant="caption" color="text.secondary" sx={{ p: 1 }}>
        Keine Änderungshistorie verfügbar
      </Typography>
    );
  }
  
  return (
    <Accordion
      expanded={expanded}
      onChange={(_, isExpanded) => setExpanded(isExpanded)}
      sx={{
        boxShadow: 'none',
        borderTop: `1px solid ${theme.palette.divider}`,
        '&:before': { display: 'none' },
        backgroundColor: 'transparent',
      }}
    >
      <AccordionSummary
        expandIcon={<ExpandMoreIcon fontSize="small" />}
        sx={{
          minHeight: 36,
          '& .MuiAccordionSummary-content': {
            margin: '4px 0',
            alignItems: 'center',
          },
        }}
      >
        <Stack direction="row" spacing={1} alignItems="center">
          <HistoryIcon fontSize="small" color="action" />
          <Typography variant="caption" color="text.secondary">
            {lastChange 
              ? `Zuletzt geändert ${lastChange.time} von ${lastChange.user}`
              : 'Keine Änderungen'}
          </Typography>
          {lastChange && (
            <Chip
              label={formatActionLabel(lastChange.action)}
              size="small"
              color={getActionColor(lastChange.action)}
              sx={{ height: 18, fontSize: '0.7rem' }}
            />
          )}
        </Stack>
      </AccordionSummary>
      
      <AccordionDetails sx={{ pt: 0, pb: 2 }}>
        {compact ? (
          // Compact view - simple list
          <Stack spacing={1}>
            {auditEntries.content.slice(0, maxEntries).map((entry: AuditEntry) => (
              <Stack 
                key={entry.id} 
                direction="row" 
                spacing={1} 
                alignItems="flex-start"
                sx={{
                  p: 1,
                  borderRadius: 1,
                  backgroundColor: alpha(theme.palette.background.default, 0.5),
                  '&:hover': {
                    backgroundColor: alpha(theme.palette.primary.main, 0.05),
                  },
                }}
              >
                {getActionIcon(entry.eventType)}
                <Box sx={{ flex: 1, minWidth: 0 }}>
                  <Typography variant="caption" sx={{ fontWeight: 500 }}>
                    {formatActionLabel(entry.eventType)}
                  </Typography>
                  {entry.changes && (
                    <Typography 
                      variant="caption" 
                      color="text.secondary"
                      sx={{ 
                        display: 'block',
                        overflow: 'hidden',
                        textOverflow: 'ellipsis',
                        whiteSpace: 'nowrap',
                      }}
                    >
                      {formatChangeDetails(entry.changes)}
                    </Typography>
                  )}
                  <Typography variant="caption" color="text.secondary">
                    {format(new Date(entry.timestamp), 'dd.MM.yyyy HH:mm', { locale: de })}
                    {' • '}
                    {entry.userName || 'System'}
                  </Typography>
                </Box>
              </Stack>
            ))}
          </Stack>
        ) : (
          // Detailed view - timeline
          <Timeline position="right" sx={{ p: 0, m: 0 }}>
            {auditEntries.content.slice(0, maxEntries).map((entry: AuditEntry, index: number) => (
              <TimelineItem key={entry.id} sx={{ minHeight: 60 }}>
                <TimelineOppositeContent sx={{ flex: 0.3, px: 1 }}>
                  <Typography variant="caption" color="text.secondary">
                    {format(new Date(entry.timestamp), 'HH:mm')}
                  </Typography>
                  <Typography variant="caption" color="text.secondary" display="block">
                    {format(new Date(entry.timestamp), 'dd.MM.yy')}
                  </Typography>
                </TimelineOppositeContent>
                
                <TimelineSeparator>
                  <TimelineDot color={getActionColor(entry.eventType)}>
                    {getActionIcon(entry.eventType)}
                  </TimelineDot>
                  {index < auditEntries.content.length - 1 && <TimelineConnector />}
                </TimelineSeparator>
                
                <TimelineContent sx={{ px: 1 }}>
                  <Typography variant="body2" sx={{ fontWeight: 500 }}>
                    {formatActionLabel(entry.eventType)}
                  </Typography>
                  {entry.changes && (
                    <Typography variant="caption" color="text.secondary">
                      {formatChangeDetails(entry.changes)}
                    </Typography>
                  )}
                  <Typography variant="caption" color="text.secondary" display="block">
                    von {entry.userName || 'System'}
                  </Typography>
                </TimelineContent>
              </TimelineItem>
            ))}
          </Timeline>
        )}
        
        {/* Show more button */}
        {onShowMore && auditEntries.totalElements > maxEntries && (
          <Box sx={{ mt: 2, textAlign: 'center' }}>
            <Button
              size="small"
              onClick={onShowMore}
              startIcon={<HistoryIcon />}
            >
              Vollständige Historie anzeigen ({auditEntries.totalElements} Einträge)
            </Button>
          </Box>
        )}
      </AccordionDetails>
    </Accordion>
  );
}