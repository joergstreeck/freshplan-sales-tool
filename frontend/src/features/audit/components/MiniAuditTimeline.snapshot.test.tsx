/**
 * Snapshot Tests for MiniAuditTimeline Component
 *
 * Tests visual consistency of the audit timeline component.
 *
 * @module MiniAuditTimeline.snapshot.test
 * @since FC-005 PR4
 */

import { describe, it, expect, vi } from 'vitest';
import { render, act, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import React from 'react';
import { MiniAuditTimeline } from './MiniAuditTimeline';
import type { AuditEntry } from '../types/audit.types';
import { auditApi } from '../services/auditApi';

// Mock the API
vi.mock('../services/auditApi', () => ({
  auditApi: {
    getEntityAuditTrail: vi.fn(() =>
      Promise.resolve({
        entries: [
          {
            id: '1',
            entityType: 'CUSTOMER',
            entityId: '1',
            action: 'UPDATE',
            fieldName: 'status',
            oldValue: 'INACTIVE',
            newValue: 'ACTIVE',
            userId: 'user-1',
            userName: 'Max Mustermann',
            timestamp: '2024-01-15T10:30:00Z',
            details: 'Status wurde auf aktiv gesetzt',
          },
          {
            id: '2',
            entityType: 'CUSTOMER',
            entityId: '1',
            action: 'UPDATE',
            fieldName: 'revenue',
            oldValue: '500000',
            newValue: '750000',
            userId: 'user-2',
            userName: 'Anna Schmidt',
            timestamp: '2024-01-14T15:45:00Z',
            details: 'Umsatz aktualisiert',
          },
          {
            id: '3',
            entityType: 'CUSTOMER',
            entityId: '1',
            action: 'CREATE',
            userId: 'user-3',
            userName: 'Peter Weber',
            timestamp: '2024-01-10T09:00:00Z',
            details: 'Kunde angelegt',
          },
        ] as AuditEntry[],
        totalCount: 3,
        hasMore: false,
      })
    ),
    getLatestChange: vi.fn(() =>
      Promise.resolve({
        id: '1',
        entityType: 'CUSTOMER',
        entityId: '1',
        action: 'UPDATE',
        fieldName: 'status',
        oldValue: 'INACTIVE',
        newValue: 'ACTIVE',
        userId: 'user-1',
        userName: 'Max Mustermann',
        timestamp: '2024-01-15T10:30:00Z',
      } as AuditEntry)
    ),
  },
}));

// Mock Auth Context
vi.mock('../../../contexts/AuthContext', () => ({
  useAuth: () => ({
    user: { id: 'test-user', name: 'Test User', roles: ['admin'] },
    isAuthenticated: true,
  }),
}));

// Mock date-fns for consistent snapshots
vi.mock('date-fns', () => ({
  formatDistanceToNow: () => 'vor 2 Stunden',
  format: () => '15.01.2024 10:30',
}));

// Test wrapper
const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: { retry: false, staleTime: Infinity },
      mutations: { retry: false },
    },
  });
  const theme = createTheme();

  return ({ children }: { children: React.ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </QueryClientProvider>
  );
};

describe('MiniAuditTimeline - Snapshot Tests', () => {
  const defaultProps = {
    entityType: 'CUSTOMER',
    entityId: '1',
  };

  beforeEach(() => {
    // Reset mocks before each test
    vi.clearAllMocks();

    // Set default mock response
    vi.mocked(auditApi.getEntityAuditTrail).mockResolvedValue({
      entries: [
        {
          id: '1',
          entityType: 'CUSTOMER',
          entityId: '1',
          action: 'UPDATE',
          fieldName: 'status',
          oldValue: 'INACTIVE',
          newValue: 'ACTIVE',
          userId: 'user-1',
          userName: 'Max Mustermann',
          timestamp: '2024-01-15T10:30:00Z',
          details: 'Status wurde auf aktiv gesetzt',
        },
      ] as AuditEntry[],
      totalCount: 1,
      hasMore: false,
    });
  });

  it('should match snapshot - compact mode (default)', async () => {
    let renderResult;

    await act(async () => {
      renderResult = render(<MiniAuditTimeline {...defaultProps} />, {
        wrapper: createWrapper(),
      });
    });

    // Wait for data to load (or empty state)
    await waitFor(() => {
      expect(renderResult.container.textContent.length).toBeGreaterThan(0);
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('compact-mode');
  });

  it('should match snapshot - expanded mode', async () => {
    let renderResult;

    await act(async () => {
      renderResult = render(<MiniAuditTimeline {...defaultProps} compact={false} />, {
        wrapper: createWrapper(),
      });
    });

    await waitFor(() => {
      expect(renderResult.container.textContent.length).toBeGreaterThan(0);
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('expanded-mode');
  });

  it('should match snapshot - with details', async () => {
    let renderResult;

    await act(async () => {
      renderResult = render(
        <MiniAuditTimeline {...defaultProps} showDetails={true} compact={false} />,
        { wrapper: createWrapper() }
      );
    });

    await waitFor(() => {
      expect(renderResult.container.textContent.length).toBeGreaterThan(0);
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('with-details');
  });

  it('should match snapshot - limited entries', async () => {
    let renderResult;

    await act(async () => {
      renderResult = render(
        <MiniAuditTimeline {...defaultProps} maxEntries={2} compact={false} />,
        { wrapper: createWrapper() }
      );
    });

    await waitFor(() => {
      expect(renderResult.container.textContent.length).toBeGreaterThan(0);
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('limited-entries');
  });

  it('should match snapshot - loading state', () => {
    // Mock loading state
    vi.mocked(auditApi.getEntityAuditTrail).mockImplementation(
      () => new Promise(() => {}) // Never resolves
    );

    const { container } = render(<MiniAuditTimeline {...defaultProps} />, {
      wrapper: createWrapper(),
    });

    expect(container.firstChild).toMatchSnapshot('loading-state');
  });

  it('should match snapshot - empty state', async () => {
    // Mock empty response
    vi.mocked(auditApi.getEntityAuditTrail).mockResolvedValueOnce({
      entries: [],
      totalCount: 0,
      hasMore: false,
    });

    const { container } = render(<MiniAuditTimeline {...defaultProps} />, {
      wrapper: createWrapper(),
    });

    await vi.waitFor(() => {
      expect(container.textContent).toContain('Keine Ã„nderungshistorie');
    });

    expect(container.firstChild).toMatchSnapshot('empty-state');
  });

  it('should match snapshot - error state', async () => {
    // Mock error
    vi.mocked(auditApi.getEntityAuditTrail).mockRejectedValueOnce(
      new Error('Failed to load audit history')
    );

    let renderResult;

    await act(async () => {
      renderResult = render(<MiniAuditTimeline {...defaultProps} />, {
        wrapper: createWrapper(),
      });
    });

    await waitFor(() => {
      expect(renderResult.container.textContent).toContain(
        'Audit-Historie konnte nicht geladen werden'
      );
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('error-state');
  });

  it('should match snapshot - with show more button', async () => {
    // Mock response with more entries available
    vi.mocked(auditApi.getEntityAuditTrail).mockResolvedValueOnce({
      entries: Array(5)
        .fill(null)
        .map((_, i) => ({
          id: `${i}`,
          entityType: 'CUSTOMER',
          entityId: '1',
          action: 'UPDATE',
          userId: 'user-1',
          userName: 'Test User',
          timestamp: new Date().toISOString(),
        })),
      totalCount: 20,
      hasMore: true,
    });

    let renderResult;

    await act(async () => {
      renderResult = render(<MiniAuditTimeline {...defaultProps} onShowMore={vi.fn()} />, {
        wrapper: createWrapper(),
      });
    });

    await waitFor(() => {
      expect(renderResult.container.textContent.length).toBeGreaterThan(0);
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('with-show-more');
  });

  it('should match snapshot - different entity types', async () => {
    let renderResult;

    await act(async () => {
      renderResult = render(<MiniAuditTimeline entityType="CONTACT" entityId="contact-1" />, {
        wrapper: createWrapper(),
      });
    });

    await waitFor(() => {
      expect(renderResult.container.textContent.length).toBeGreaterThan(0);
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('contact-entity');
  });

  it('should match snapshot - dark mode', async () => {
    const darkTheme = createTheme({
      palette: {
        mode: 'dark',
      },
    });

    const DarkWrapper = ({ children }: { children: React.ReactNode }) => (
      <QueryClientProvider client={new QueryClient()}>
        <ThemeProvider theme={darkTheme}>{children}</ThemeProvider>
      </QueryClientProvider>
    );

    let renderResult;

    await act(async () => {
      renderResult = render(<MiniAuditTimeline {...defaultProps} />, { wrapper: DarkWrapper });
    });

    await waitFor(() => {
      expect(renderResult.container.textContent.length).toBeGreaterThan(0);
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('dark-mode');
  });

  it('should match snapshot - all action types', async () => {
    // Mock with different action types
    vi.mocked(auditApi.getEntityAuditTrail).mockResolvedValueOnce({
      entries: [
        {
          id: '1',
          entityType: 'CUSTOMER',
          entityId: '1',
          action: 'CREATE',
          userId: 'user-1',
          userName: 'User 1',
          timestamp: '2024-01-15T10:00:00Z',
        },
        {
          id: '2',
          entityType: 'CUSTOMER',
          entityId: '1',
          action: 'UPDATE',
          fieldName: 'name',
          oldValue: 'Old Name',
          newValue: 'New Name',
          userId: 'user-2',
          userName: 'User 2',
          timestamp: '2024-01-15T11:00:00Z',
        },
        {
          id: '3',
          entityType: 'CUSTOMER',
          entityId: '1',
          action: 'DELETE',
          userId: 'user-3',
          userName: 'User 3',
          timestamp: '2024-01-15T12:00:00Z',
        },
      ],
      totalCount: 3,
      hasMore: false,
    });

    let renderResult;

    await act(async () => {
      renderResult = render(
        <MiniAuditTimeline {...defaultProps} compact={false} showDetails={true} />,
        { wrapper: createWrapper() }
      );
    });

    await waitFor(() => {
      expect(renderResult.container.textContent.length).toBeGreaterThan(0);
    });

    expect(renderResult.container.firstChild).toMatchSnapshot('all-action-types');
  });
});
