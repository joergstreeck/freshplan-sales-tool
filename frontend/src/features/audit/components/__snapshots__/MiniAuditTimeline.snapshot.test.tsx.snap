// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - all action types > all-action-types 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1frpshr-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - compact mode (default) > compact-mode 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1frpshr-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - dark mode > dark-mode 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1px7uz0-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - different entity types > contact-entity 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1frpshr-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - empty state > empty-state 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1frpshr-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - error state > error-state 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-1h5lil8-MuiPaper-root-MuiAlert-root"
  role="alert"
  style="--Paper-shadow: none;"
>
  <div
    class="MuiAlert-icon css-vab54s-MuiAlert-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
      data-testid="ErrorOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    class="MuiAlert-message css-zioonp-MuiAlert-message"
  >
    Audit-Historie konnte nicht geladen werden
  </div>
</div>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - expanded mode > expanded-mode 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1frpshr-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - limited entries > limited-entries 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1frpshr-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - loading state > loading-state 1`] = `
<div
  class="MuiBox-root css-hpgf8j"
>
  <span
    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
    style="width: 80%; height: 20px;"
  />
</div>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - with details > with-details 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1frpshr-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;

exports[`MiniAuditTimeline - Snapshot Tests > should match snapshot - with show more button > with-show-more 1`] = `
<span
  class="MuiTypography-root MuiTypography-caption css-1frpshr-MuiTypography-root"
>
  Keine Änderungshistorie verfügbar
</span>
`;
